### PR TITLE
replacing iris data from examples

### LINKED ISSUE #10157: Replacing Iris data with penguin data
So given recent events there has been a lot of discussion about the Iris data, which is a standard for demonstrating classification and other statistical techniques. The problem with it is that was first published by Ronald Fisher (a Eugenicist) in the Annals of Eugenics and I don't think I need to point out why that history is extremely problematic for a dataset that has been used for classification problems. It's also incredibly overused and many people will roll their eyes just seeing it.

Thankfully there is a new dataset a lot of people seem to be coalescing around, specifically this open source dataset about penguins: https://github.com/allisonhorst/palmerpenguins, which has several advantages:

* It wasn't popularized by a eugenicist
* It has more interesting dimensions to explore
* It's about penguins 🐧 🐧 🐧 

I therefore propose we replace the Iris data in sampledata and all the associated examples with equivalent examples using the penguin data. Libraries like seaborn have already taken this step and I think we should too.

### PR DESCRIPTION
**Pull request by mosc9575**
I removed the iris data from all examples except https://docs.bokeh.org/en/latest/docs/examples/plotting/iris.html, because is has already a comment on it.

- [ ] issues: fixes #10157

### CODE REVIEW

**Comment by bryevdv** [comment]
Oh wow, thanks @mosc9575 I keep forgetting there were some stragglers to fix up. Looks like this branch accidentally picked up the js callbacks docs commit, maybe a rebase on `branch-3.4` is needed?

**Review by bryevdv — APPROVED** [review]

File: examples/basic/layouts/words_and_plots.py
Comment (mosc9575) [inline examples/basic/layouts/words_and_plots.py]:
```suggestion
    colormap = {'Adelie': 'red', 'Chinstrap': 'green', 'Gentoo': 'blue'}
```

**Comment by bryevdv** [comment]
@mosc9575 I think you should be able to push the merge button, correct? If so let's switch to our usual "self-merge after approval and tests" approach for your PRs  (and if you aren't able to push  merge button, let's figure out why)

**Comment by mosc9575** [comment]
@bryevdv Yes, I can merge and I will go for it. Thanks a lot for your trust.

**Pull request merged**
